src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionErrorTest.java

```diff
@@ -0,0 +1,160 @@
+import static io.pinecone.helpers.IndexManager.createCollection;
+import static io.pinecone.helpers.IndexManager.waitUntilIndexIsReady;
+import static io.pinecone.helpers.BuildUpsertRequest.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CollectionErrorTest {
+    private static final String apiKey = System.getenv("PINECONE_API_KEY");
+    private static final String environment = System.getenv("PINECONE_ENVIRONMENT");
+    private static final String indexName = RandomStringBuilder.build("collection-error-test", 8);
+    private static final String collectionName = RandomStringBuilder.build("reusable-coll", 8);
+    private static final ArrayList<String> indexes = new ArrayList<>();
+    private static final ArrayList<String> collections = new ArrayList<>();
+    private static final List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+    private static final int dimension = 4;
+    private static PineconeControlPlaneClient controlPlaneClient;
+
+    @BeforeAll
+    public static void setUpIndexAndCollection() throws InterruptedException {
+        controlPlaneClient = new PineconeControlPlaneClient(apiKey);
+        CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
+        CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
+        PineconeConnection dataPlaneConnection = createNewIndexAndConnect(controlPlaneClient, indexName, dimension, IndexMetric.COSINE, spec);
+        VectorServiceGrpc.VectorServiceBlockingStub blockingStub = dataPlaneConnection.getBlockingStub();
+        indexes.add(indexName);
+
+        // Upsert vectors to index and sleep for freshness
+        blockingStub.upsert(buildRequiredUpsertRequestByDimension(upsertIds, dimension, ""));
+        dataPlaneConnection.close();
+        Thread.sleep(3500);
+
+        // Create collection from index
+        createCollection(controlPlaneClient, collectionName, indexName, true);
+        collections.add(collectionName);
+    }
+
+    @AfterAll
+    public static void cleanUp() throws InterruptedException {
+        // wait for things to settle before cleanup...
+        Thread.sleep(2500);
+        for (String index : indexes) {
+            controlPlaneClient.deleteIndex(index);
+        }
+        for (String collection : collections) {
+            controlPlaneClient.deleteCollection(collection);
+        }
+        Thread.sleep(2500);
+    }
+
+    @Test
+    public void testCreateCollectionFromInvalidIndex() {
+        try {
+            CreateCollectionRequest createCollectionRequest = new CreateCollectionRequest().name(RandomStringBuilder.build("coll1", 8)).source("invalid-index");
+            controlPlaneClient.createCollection(createCollectionRequest);
+        } catch (PineconeException exception) {
+            System.out.println("Exception: " + exception.getMessage());
```

Comment:
These print statements can be removed right?

Reply:
I actually added them because they're helpful for debugging if the test fails. With only the assertion, if the `exception` doesn't contain what we expect we don't see the actual error message in the output, so I feel like this is probably fine to leave in for when/if these tests fail and we need to debug.